packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -1085,6 +1085,48 @@ mixin SchedulerBinding on BindingBase {
     }
   }
 
+  final Map<dynamic, DartPerformanceMode> _performanceModes = <Object, DartPerformanceMode>{};
+
+  /// Request a specific [DartPerformanceMode]. `requestor` is the handle to
+  /// component making the request, typically the component itself. Returns
+  /// `true` is the request was successfully made to the engine, `false` otherwise.
+  /// Even if the result is `true`, the engine may choose to ignore the request or
+  /// the performance mode may not be guaranteed to be the one requested.
+  ///
+  /// If conflicting requests are made, only the first request will be honored.
+  bool createPerformanceModeRequest(Object requestor, DartPerformanceMode mode) {
```

Comment:
In other APIs we often make this return a handle object that you need to dispose to undo the request if you no longer need it.